react/src/hooks/useTableFilterCache.tsx

```diff
@@ -0,0 +1,48 @@
+import { Column, Filter } from "@material-table/core";
+
+/**
+ * Return functions for handling table filter caching.
+ *
+ * Instead of loading saved filters on mount via side effect,
+ * use `setInitialFilters` to set the default filter values
+ * for all columns before providing them to material-table as a prop.
+ */
+export function useTableFilterCache<RowData extends object>(cacheKey: string) {
```

Comment:
This may not necessarily need to be a hook. The React docs are kind of vague on this point, but if 'the rules of hooks' don't have to be followed (b/c the function isn't itself wrapping hooks), then 2 plain functions might suffice (and avoid eslint errors that target hook-specific rules).

Reply:
Yeah I considered that this doesn't necessarily need to be a hook, but I think it still works as a convenient way to bundle both functions together, since both are required for full functionality. The `cacheKey` and typing information also get passed down to both functions, preventing us from having to write that info twice if the functions were made separate.